docs/using/troubleshooting.md

```diff
@@ -0,0 +1,310 @@
+        -e NB_USER=callisto \
+        -e NB_UID=1234 \
+        -e NB_GID=1234 \
+        -e CHOWN_HOME=yes \
+        -e CHOWN_HOME_OPTS="-R" \
+        -w "/home/${NB_USER}" \
+        -v "${PWD}"/test:/home/callisto/work \
+        jupyter/minimal-notebook
+
+    # Updated the jovyan user:
+    # - username: jovyan       -> callisto
+    # - home dir: /home/jovyan -> /home/callisto
+    # Update callisto UID:GID to 1234:1234
+    # Attempting to copy /home/jovyan to /home/callisto...
+    # Success!
+    # Ensuring /home/callisto is owned by 1234:1234
+    # Running as callisto: bash
+   ```
+
+   where:
+
+   - `-e NB_USER=callisto`: will create a new user `callisto` and automatically add it to the `users` group (does not delete jovyan)
+   - `-e NB_UID=1234` and `-e NB_GID=1234`: will set the `UID` and `GID` of the new user (`callisto`) to `1234`
+   - `-e CHOWN_HOME_OPTS="-R"` and `-e CHOWN_HOME=yes`: ensure that the new user is the owner of the `/home` directory and subdirectories
+     (setting `CHOWN_HOME_OPTS="-R` will ensure this change is applied recursively)
+   - `-w "/home/${NB_USER}"` sets the working directory to be the new user's home
+
+   ```{admonition} Additional notes
+    In the example above, the `-v` flag is used to mount the local volume onto the new user's `/home` directory.
+
+    However, if you are mounting a volume elsewhere, you also need to use the `-e CHOWN_EXTRA=<some-dir>` flag to avoid any permission
+    issues (see the section [Permission denied when mounting volumes](../using/troubleshooting.html#permission-denied-when-mounting-volumes) in this page).
+   ```
+
+2. **Dynamically assign the user ID and GID**
+
+   The above case ensures that the `/home` directory is owned by the a newly created user with an specific `UID` and `GID`, but if you want to assign the `UID` and `GID`
+   of the new user dynamically you can make the following adjustments:
+
+   ```bash
+   docker run -it --rm \
+       -p 8888:8888 \
+       --user root \
+       -e NB_USER=callisto \
+       -e NB_UID="$(id -u)" \
+       -e NB_GID="$(id -g)"  \
+       -e CHOWN_HOME=yes \
+       -e CHOWN_HOME_OPTS="-R" \
+       -w "/home/${NB_USER}" \
+       -v "${PWD}"/test:/home/callisto/work \
+       jupyter/minimal-notebook
+   ```
+
+   where:
+
+   - `"$(id -u)" and "$(id -g)"` will dynamically assign the `UID` and `GID` of the user executing the `docker run` command to the new user (`callisto`)
+
+## Additional tips and troubleshooting commands for permission-related errors
+
+- Pass absolute paths to the `-v` flag:
+
+  ```bash
+  -v "${PWD}"/<my-vol>:/home/jovyan/work
+  ```
+
+  This example uses the syntax `$(PWD)`, which is replaced with the full path to the current directory at runtime. The destination
+  path should also be an absolute path starting with a `/` such as `home/jovyan/work`.
+
+- You might want to consider using the Docker native `--user <UID>` and `--group-add users` flags instead of `-e NB_GID` and `-e NB_UID`:
```

Comment:
```suggestion
- You might want to consider using the Docker native `--user <UID>` and `--group-add users` flags instead of `-e NB_UID` and `-e NB_GID`:
```